src/tests/ftest/deployment/critical_integration.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import time
+
+from general_utils import run_command, DaosTestError
+from ior_test_base import IorTestBase
+from apricot import TestWithServers
+from command_utils_base import CommandFailure
+
+
+class CriticalIntegration(TestWithServers):
+    """Test Class Description: Verify the basic integration of
+                               the server nodes with available
+                               framework and amongst themselves.
+    :avocado: recursive
+    """
+
+    #TO-DO
+    #Provision all daos nodes using provisioning tool provided by HPCM
+
+    def test_passwdlessssh_versioncheck(self):
+        # pylint: disable=protected-access
+        """
+        Test Description: Verify passwordless ssh amongst the server
+                          server nodes available and verify all server
+                          and client nodes have same daos versions.
+        :avocado: tags=all,deployment,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=criticalintegration,passwdlessssh_versiocheck
+        """
+
+        check_remote_root_access = self.params.get("check_remote_root_access", "/run/*")
+        daos_server_version_list = []
+        dmg_version_list = []
+        for host in self.hostlist_servers:
+            daos_server_cmd = ("ssh -oNumberOfPasswordPrompts=0 {}"
+                               " 'daos_server version'".format(host))
+            remote_root_access = ("ssh -oNumberOfPasswordPrompts=0 root@{}"
+                                  " 'echo hello'".format(host))
+            command_for_inter_node = ("clush --nostdin -S -w {}"
+                                      " 'echo hello'".format(','.join(self.hostlist_servers)))
+            try:
+                out = run_command(daos_server_cmd)
+                daos_server_version_list.append(out.stdout.split(b' ')[3])
+                if check_remote_root_access:
+                    run_command(remote_root_access)
+                IorTestBase._execute_command(self, command_for_inter_node, hosts=[host])
+            except (DaosTestError, CommandFailure) as error:
+                self.fail("Ssh check Failed.\n {}".format(error))
+
```

Comment:
Should each of the above command be in a separate try?    Or add more logging info to determine what failed?  
Does the fail force the test to exit on first failure?   Maybe change it to gather all the nodes that failed the checks and report at the end of the test if anything failed.   That way you have all the failing nodes but can continue with the rest of the test

Reply:
Implemented your second suggestion to continue on error and fail later. Regarding other suggestion, I believe that information will be evident when printing the 'error' message anyways.